subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildConfigurationAttributesResolveIntegrationTest.groovy

```diff
@@ -478,7 +478,6 @@ class CompositeBuildConfigurationAttributesResolveIntegrationTest extends Abstra
         type         | freeValue                      | paidValue
         'SomeEnum'   | 'SomeEnum.free'                | 'SomeEnum.paid'
         'Thing'      | 'objects.named(Thing, "free")' | 'objects.named(Thing, "paid")'
-        'OtherThing' | 'new OtherThing(name: "free")' | 'new OtherThing(name: "paid")'
```

Comment:
💭 Having a field in a Named implementation class is not allowed, not sure why this worked previously.

```
* What went wrong:
Could not determine the dependencies of task ':a:checkFree'.
> Could not resolve all dependencies for configuration ':a:_compileFree'.
   > Could not select a variant of project :includedBuild that matches the consumer attributes.
      > Could not create an instance of type OtherThing.
         > Type OtherThing is not a valid Named implementation class:
           - Field name is not valid: A Named implementation class must not define any instance fields.
```